### DISCUSSION TITLE
Using primary_ip instead of resolved IP from DNS

### DISCUSSION THREAD

**Discussion by obol89** [netbox-community/ansible_modules#1010] [General]
Hi All,

I'm using `nb_inventory` as our dynamic inventory source.
It looks like ansible is trying to use `primary_ip` from netbox when reaching the host, instead of using DNS name and get IP from our DNS server.
We changed the name of one of our hosts recently, let's say from server1 to server2. It had interface `eth0` with assigned address `10.9.61.82/24` set as a `primary_ip.` 
nslookup was resolving a different address that is `192.168.249.67/24,` and it is correct according to our DNS server.
I was able to reach this host on its correct address with Ansible, after I've removed the `primary_ip` from Netbox.

I was trying to play with various parameters, but nothing helped. How to avoid situation like that? Is this normal behaviour or some misconfiguration on our side?

**Comment by sc68cal** [comment]
The inventory plugin supports two options:

* `ansible_host_dns_name`
* `dns_name`

Consult the documentation for their usage, but in general the inventory plugin depends on `primary_ip` for setting the `ansible_host` value, so you need to ensure that your NetBox data is up to date and correct. [Both settings rely on looking up the NetBox IPAddress object that has been set as the `primary_ip` for a device, and then looking at the `dns_name` value of the NetBox object.](https://github.com/netbox-community/ansible_modules/blob/devel/plugins/inventory/nb_inventory.py#L963). 

If you are relying on DNS instead of NetBox to manage the IP address that is used to access a device, you may not want to use the inventory plugin, and instead use the query plugin.

I have not used the inventory plugin extensively and have more experience using the query plugin, so I could be mistaken, but this is based on reading the code over coffee.

**Comment by wols** [comment]
Today testet:

`ansible-inventory [core 2.15.4]`

```
# netbox_inventory.yml
plugin: netbox.netbox.nb_inventory
ansible_host_dns_name: False
```
`ansible-inventory -i netbox_inventory.yml --list`
```
"APU2C4": {
  "ansible_host": "192.168.1.10",
```
---
```
# netbox_inventory.yml
plugin: netbox.netbox.nb_inventory
ansible_host_dns_name: True
```
```
"APU2C4": {
  "ansible_host": "apu.my.domain",
```
